R/aes-evaluation.r

```diff
@@ -93,31 +93,38 @@ is_dotted_var <- function(x) {
 }
 
 # Determine if aesthetic is calculated
-is_calculated_aes <- function(aesthetics) {
-  vapply(aesthetics, is_calculated, logical(1), USE.NAMES = FALSE)
+is_calculated_aes <- function(aesthetics, warn = FALSE) {
+  vapply(aesthetics, is_calculated, warn = warn, logical(1), USE.NAMES = FALSE)
 }
 is_scaled_aes <- function(aesthetics) {
   vapply(aesthetics, is_scaled, logical(1), USE.NAMES = FALSE)
 }
 is_staged_aes <- function(aesthetics) {
   vapply(aesthetics, is_staged, logical(1), USE.NAMES = FALSE)
 }
-is_calculated <- function(x) {
+is_calculated <- function(x, warn = FALSE) {
   if (is_call(get_expr(x), "after_stat")) {
     return(TRUE)
   }
   # Support of old recursive behaviour
   if (is.atomic(x)) {
     FALSE
   } else if (is.symbol(x)) {
-    is_dotted_var(as.character(x))
+    res <- is_dotted_var(as.character(x))
+    if (res && warn) {
+      lifecycle::deprecate_warn("3.4.0", I("The dot-dot notation (`..var..`)"), "after_stat()")
```

Comment:
Could we interpolate the variable name in to `..{{var}}..` and `after_stat({var}})`? (assuming `var <- as.character(x))`)

Reply:
Do you mean this is supposed to work...?

``` r
var <- "foo"
lifecycle::deprecate_warn("3.4.0", "a()", "b({{var}})")
#> Error in `as_string()`:
#> ! Can't convert a call to a string.

#> Backtrace:
#>     ▆
#>  1. └─lifecycle::deprecate_warn("3.4.0", "a()", "b({{var}})")
#>  2.   ├─id %||% msg
#>  3.   └─lifecycle:::lifecycle_message(...)
#>  4.     └─lifecycle:::spec(with, NULL, signaller)
#>  5.       └─lifecycle:::spec_arg(what$call, signaller = signaller)
#>  6.         └─rlang::as_string(node_car(arg))
#>  7.           └─rlang:::abort_coercion(x, "a string")
#>  8.             └─rlang::abort(msg, call = call)
```

<sup>Created on 2022-08-25 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Oh sorry, you’ll need to glue/paste yourself.

Reply:
I see. Thanks for the suggestion anyway!